pkg/listener/job_processor.go

```diff
@@ -227,16 +261,18 @@ func (p *JobProcessor) disconnect() {
 
 func (p *JobProcessor) Shutdown(reason string, code int) {
 	p.disconnect()
-	p.executeShutdownHook()
-	log.Info(reason)
-	log.Info("Shutting down... Good bye!")
+	p.executeShutdownHook(reason)
+	log.Infof("Agent shutting down due to: %s", reason)
 	os.Exit(code)
 }
 
-func (p *JobProcessor) executeShutdownHook() {
+func (p *JobProcessor) executeShutdownHook(reason string) {
```

Comment:
Use typed enums. It will reduce the chance of bugs in the long run.

https://www.sohamkamani.com/golang/enums/#enum-types

```

```